psdisk/psdisk.c

```diff
@@ -359,227 +393,159 @@ static int psdisk_updatePartsList(ptable_partition_t *pHeaders, uint32_t pCnt)
 
 static int psdisk_updateImg(void)
 {
-	uint32_t pCnt;
-	ptable_partition_t *pHeaders;
+	ptable_t *ptable;
+	int err;
 
-	if ((pHeaders = ptable_readPartitions(&pCnt, &psdisk_common.mem)) == NULL) {
-		fprintf(stderr,"The file contains incorrect partition table.\n");
+	ptable = psdisk_readImg();
+	if (ptable == NULL) {
+		fprintf(stderr, "The file contains incorrect partition table.\n");
 		return -1;
 	}
 
-	if (psdisk_updatePartsList(pHeaders, pCnt) < 0) {
-		free(pHeaders);
+	err = psdisk_updatePartsList(ptable);
+	free(ptable);
+	if (err < 0) {
 		return -1;
 	}
 
 	if (psdisk_createImg() < 0) {
-		free(pHeaders);
 		return -1;
 	}
 
-	free(pHeaders);
-
 	return 0;
 }
 
 
 /* Parsing data from user */
 
-static int psdisk_parseToRm(void)
+static int psdisk_parseToRm(const char *arg)
```

Comment:
Might be `name` instead `arg`.
```suggestion
static int psdisk_parseToRm(const char *name)
```